docker/Dockerfile_cuda_stack

```diff
@@ -0,0 +1,336 @@
+from ubuntu:14.04
+
+ARG NPROC=12
+
+RUN apt-get update && apt-get install -y \
+        build-essential \
+        gfortran \
+        openmpi-bin \
+        libopenmpi-dev \
+        curl \
+        cmake \
+        git \
+        wget \
+        vim \
+        emacs \
+        autoconf \
+        lcov \
+        valgrind \
+        ccache \
+        cppcheck \
+        libssl-dev \
+        libpng-dev \
+        libfreetype6-dev \
+        libxft-dev \
+        libsqlite3-dev \
+        libbz2-dev \
+        libatlas-base-dev \
+        zlib1g-dev \
+        python2.7-dev \
+        autogen \
+        autoconf \
+        libtool \
+        && \
+    echo "deb http://apt.llvm.org/trusty/ llvm-toolchain-trusty-3.9 main" > /etc/apt/sources.list.d/llvm.list && \
+    wget -q -O - http://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add - && \
+    apt-get update && apt-get install -y \
+        clang-3.9 \
+        clang-format-3.9 \
+        && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# set up cuda
+LABEL maintainer "NVIDIA CORPORATION <cudatools@nvidia.com>"
+
+LABEL com.nvidia.volumes.needed="nvidia_driver"
+
+RUN NVIDIA_GPGKEY_SUM=d1be581509378368edeec8c1eb2958702feedf3bc3d17011adbf24efacce4ab5 && \
```

Comment:
Why are you doing this? It would be a lot simpler to use nvidia's container https://hub.docker.com/r/nvidia/cuda/ and according to the comments, they have ubuntu 16.04 images